### ISSUE TITLE
Verify the set of machines that have docker installed and update labels

### ISSUE THREAD

**Issue by smlambert** [AdoptOpenJDK/openjdk-infrastructure#1044]
We currently have 5 machines labelled with docker for use by test: https://ci.adoptopenjdk.net/label/ci.role.test&&hw.arch.x86&&sw.os.linux&&sw.tool.docker/ 

Given that docker is in the ansible playbooks, I think there are actually more machines with docker installed than are labelled with sw.tool.docker.  We should go through the set of machines and try to run the external (docker-based tests from the openjdk-tests repo) to verify which additional test machines can be used for external tests.

<img width="731" alt="Screen Shot 2019-12-16 at 10 40 34 AM" src="https://user-images.githubusercontent.com/2836948/70920393-8445cb00-1ff0-11ea-8fc9-0d9a83057d2a.png">

**Comment by sxa555** [comment]
Bear in mind several godaddy ones currently have it installed but it doesn't quite work as per https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/721

**Comment by sophia-guo** [comment]
Yes, still have the error 'The command '/bin/sh -c apt-get update 	&& apt-get -y install 	ant 	apt-transport-https 	ca-certificates 	curl 	git 	make 	unzip 	vim' returned a non-zero code: 100'

https://ci.adoptopenjdk.net/view/work%20in%20progress/job/Grinder_Sandbox/1047/console

**Comment by smlambert** [comment]
Been through the set of https://ci.adoptopenjdk.net/label/hw.arch.ppc64le/ test machines, 5/11 have docker installed (ubuntu based ones), label is present.

Going through [s390x](https://ci.adoptopenjdk.net/label/hw.arch.s390x/) now, looks like docker present on test-marist-ubuntu1604-s390x-4 so label added, will confirm others and add labels to remaining machines shortly.

**Comment by Willsparker** [comment]
I thought I'd help the effort. I used Ansible to just query all non-windows test machines with `docker --version`. (For ease, I used the superuser that is meant to be setup on all machines).
The following are the test machines that have docker installed in some capacity.
```
AWS: 
   ubuntu1804-armv8-1

GoDaddy:
  centos7-x64-1
  centos7-x64-2
  centos7-x64-3
  centos7-x64-4
  debian8-x64-1
  debian8-x64-2
  debian8-x64-3
  debian8-x64-4
  ubuntu1604-x64-1
  ubuntu1604-x64-2
  ubuntu1604-x64-3
  ubuntu1604-x64-4

Osuosl:
  ubuntu1604-ppc64le-1
  ubuntu1604-ppc64le-2
  ubuntu1604-ppc64le-3
  ubuntu1604-ppc64le-4
  ubuntu1804-ppc64le-1
  ubuntu1804-ppc64le-2

Packet:
  ubuntu1604-x64-1
  ubuntu1604-x64-2
  ubuntu1604-x64-3

Marist:
  sles12-s390x-1
  ubuntu1604-s390x-1
  ubuntu1604-s390x-2
  ubuntu1604-s390x-3
  ubuntu1604-s390x-4

Scaleway:
  ubuntu1604-x64-1

Softlayer:
  rhel74-x64-1
  ubuntu1604-x64-1
```

**Comment by sxa** [comment]
Bumping this for two reasons:
1. A number of the machines listed above, such as the GoDaddy systems, are no longer around, and many of the ubuntu1604 machines are no longer Ubuntu 16.04
2. Given that we have good capacity on Linux/aarch64, we should ensure we can run these tests on that platform as well.

**Comment by sxa** [comment]
The current list of linux/x64 test machines which do not have the ability to run docker containers are the ten machines at https://ci.adoptopenjdk.net/label/ci.role.test&&hw.arch.x86&&sw.os.linux&&!sw.tool.docker/

Of these, eight are "static docker" containers already so not suitable for running containers and must therefore not be tagged. 

I've added the tag to the ibmcloud-rhel7 one, but realistically it is preferable to run the docker jobs on the larger equinix machines that we have, which also have multiple executors which could be used. Those systems do not currently have the test tags, but (as I think we discussed elsewhere) I'd be keen to add the support for running containerised testing on them if we can define a suitable set of tags that only allows the dockerised tests to run on them. On that basis, I'll leave this open for a bit longer :-)
